pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -46,16 +47,32 @@ const (
 
 const loadBalancerProvisioningStatusActive = "ACTIVE"
 
+var lookupHost = func(host string) (addrs []string, err error) {
+	if net.ParseIP(host) != nil {
+		return []string{host}, nil
+	}
+	return net.LookupHost(host)
+}
+
 func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string, apiServerPort int) (bool, error) {
 	loadBalancerName := getLoadBalancerName(clusterName)
 	s.scope.Logger().Info("Reconciling load balancer", "name", loadBalancerName)
 
 	var fixedIPAddress string
+	var err error
+
 	switch {
 	case openStackCluster.Spec.APIServerFixedIP != "":
 		fixedIPAddress = openStackCluster.Spec.APIServerFixedIP
 	case openStackCluster.Spec.DisableAPIServerFloatingIP && openStackCluster.Spec.ControlPlaneEndpoint.IsValid():
-		fixedIPAddress = openStackCluster.Spec.ControlPlaneEndpoint.Host
+		ips, err := lookupHost(openStackCluster.Spec.ControlPlaneEndpoint.Host)
+		if err != nil {
+			return false, err
```

Comment:
Might want to wrap the error here
```suggestion
			return false, fmt.Errorf("lookup host: %w", err)
```